### PR TITLE
Parametrify crypt thingys

### PR DESCRIPTION
**Pull request by noppe**

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -80,6 +80,9 @@
   $vas_conf_libvas_site_only_servers                    = false,
   $vas_conf_libvas_use_dns_srv                          = true,
   $vas_conf_libvas_use_tcp_only                         = true,
+  $vas_conf_libvas_crypt_thingys                        = [ 'default_tgs_enctypes', 'aes256-cts-hmac-sha1-96',
+                                                            'default_tkt_enctypes', 'aes256-cts-hmac-sha1-96',
+                                                            'default_etypes',       'aes256-cts-hmac-sha1-96' ],
```
Comment (Phil-Friderici) [inline manifests/init.pp]: Please use a hash instead to make the data more structured or use independent params for each type.

File: templates/vas.conf.erb [hunk 2]
```diff
@@ -7,10 +7,9 @@
 
 [libdefaults]
  default_realm = <%= @realm.upcase %>
- default_tgs_enctypes = arcfour-hmac-md5
- default_tkt_enctypes = arcfour-hmac-md5
- default_etypes_des = des-cbc-crc
- default_etypes = arcfour-hmac-md5
+<% @vas_conf_libvas_crypt_thingys.each_slice(2) do |k, v| -%>
+  <%= k %> = <%= v %>
+<% end -%>
```
Comment (Phil-Friderici) [inline templates/vas.conf.erb]: Please avoid adding functional logic into templates.

**Comment by Phil-Friderici** [comment]
Wouldn't it be bad if something would happen to your nice little code ?
You should add spec tests to protect them against unintented changes.

File: templates/vas.conf.erb [hunk 3]
```diff
@@ -7,10 +7,9 @@
 
 [libdefaults]
  default_realm = <%= @realm.upcase %>
- default_tgs_enctypes = arcfour-hmac-md5
- default_tkt_enctypes = arcfour-hmac-md5
- default_etypes_des = des-cbc-crc
```
Comment (Phil-Friderici) [inline templates/vas.conf.erb]: what did happen to `default_etypes_des` isn't that needed anymore ?

**Comment by skalmelid** [comment]
Hi Phil!

Please look instead at https://github.com/Ericsson/puppet-module-vas/pull/128.
It contains the changes that me and Tommy actually wanted when we talked to Noppe. These changes are also deployed on 18 servers.

**Comment by noppe** [comment]
I abandoned this code due to requester rejected it.

2017-11-20 15:29 GMT+01:00 Phil Friderici <notifications@github.com>:

**Pull request closed without merging**